Boost iostream has build issue on Linux with bad dependency file name

During a build on Linux:
```
-- [hunter] BZIP2_ROOT: /media/akalsi/31c73a16-e823-44d6-913d-282db3384f8a/hunter/_Base/a54fad3/c9b4904/ae7db14/Install (ver.: 1.0.6-p1)
CMake Warning (dev) at /usr/local/share/cmake-3.3/Modules/ExternalProject.cmake:1836 (if):
  Policy CMP0054 is not set: Only interpret if() arguments as variables or
  keywords when unquoted.  Run "cmake --help-policy CMP0054" for policy
  details.  Use the cmake_policy command to set the policy and suppress this
  warning.

  Quoted variables like "x" will no longer be dereferenced when the policy is
  set to NEW.  Since the policy is not set the OLD behavior will be used.
Call Stack (most recent call first):
  /usr/local/share/cmake-3.3/Modules/ExternalProject.cmake:2356 (_ep_add_download_command)
  CMakeLists.txt:324 (ExternalProject_Add)
This warning is for project developers.  Use -Wno-dev to suppress it.

CMake Error: File /media/akalsi/31c73a16-e823-44d6-913d-282db3384f8a/hunter/_Base/Download/Hunter/0.12.5/a54fad3/Unpacked/cmake/templates/BoostBZIP2Dependency.cmake.in does not exist.
CMake Error at /media/akalsi/31c73a16-e823-44d6-913d-282db3384f8a/hunter/_Base/Download/Hunter/0.12.5/a54fad3/Unpacked/cmake/modules/hunter_install_boost_config.cmake:35 (configure_file):
  configure_file Problem configuring file
Call Stack (most recent call first):
  CMakeLists.txt:362 (hunter_install_boost_config)


-- Configuring incomplete, errors occurred!
See also "/media/akalsi/31c73a16-e823-44d6-913d-282db3384f8a/hunter/_Base/a54fad3/c9b4904/ae7db14/Build/Boost/__iostreams/Build/CMakeFiles/CMakeOutput.log".

[hunter ** FATAL ERROR **] Configure step failed (dir: /media/akalsi/31c73a16-e823-44d6-913d-282db3384f8a/hunter/_Base/a54fad3/c9b4904/ae7db14/Build/Boost/__iostreams)
[hunter ** FATAL ERROR **] [Directory:/media/akalsi/31c73a16-e823-44d6-913d-282db3384f8a/hunter/_Base/Download/Hunter/0.12.5/a54fad3/Unpacked/cmake/projects/Boost/iostreams]
```

However, a file named ```BoostBZip2Dependency.cmake.in``` exists. This seems like an issue of case mismatch.

--- a/cmake/modules/hunter_install_boost_config.cmake
+++ b/cmake/modules/hunter_install_boost_config.cmake
@@ -33,8 +33,8 @@ function(hunter_install_boost_config)
 
   if(BOOST_CONFIG_LINK_BZIP2)
     configure_file(
-        "${templates}/BoostBZIP2Dependency.cmake.in"
-        "${configs}/BoostBZIP2Dependency.cmake"
+        "${templates}/BoostBZip2Dependency.cmake.in"
+        "${configs}/BoostBZip2Dependency.cmake"
         @ONLY
     )
   endif()

